src/dotnet/Telemetry.cs

```diff
@@ -33,60 +42,73 @@ public Telemetry()
                 return;
             }
 
-            try
+            _telemetryProfile = Environment.GetEnvironmentVariable(TelemetryProfileEnvironmentVariable);
```

Comment:
Is it necessary to do this in the ctor and store the result as a private member versus reading it in InitializeTelemetry and not storing it?